node_modules/col-activities/lib/notifications/daily.js

```diff
@@ -0,0 +1,548 @@
+/**
+ * Copyright ©2015. The Regents of the University of California (Regents). All Rights Reserved.
+ *
+ * Permission to use, copy, modify, and distribute this software and its documentation
+ * for educational, research, and not-for-profit purposes, without fee and without a
+ * signed licensing agreement, is hereby granted, provided that the above copyright
+ * notice, this paragraph and the following two paragraphs appear in all copies,
+ * modifications, and distributions.
+ *
+ * Contact The Office of Technology Licensing, UC Berkeley, 2150 Shattuck Avenue,
+ * Suite 510, Berkeley, CA 94720-1620, (510) 643-7201, otl@berkeley.edu,
+ * http://ipira.berkeley.edu/industry-info for commercial licensing opportunities.
+ *
+ * IN NO EVENT SHALL REGENTS BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT, SPECIAL,
+ * INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS, ARISING OUT OF
+ * THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF REGENTS HAS BEEN ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
+ * SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF ANY, PROVIDED HEREUNDER IS PROVIDED
+ * "AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
+ * ENHANCEMENTS, OR MODIFICATIONS.
+ */
+
+var _ = require('lodash');
+var async = require('async');
+var moment = require('moment-timezone');
+
+var ActivitiesAPI = require('col-activities');
+var CourseAPI = require('col-course');
+var CollabosphereConstants = require('col-core/lib/constants');
+var DB = require('col-core/lib/db');
+var EmailUtil = require('col-core/lib/email');
+var log = require('col-core/lib/logger')('col-activities/notifications');
+var UsersAPI = require('col-users');
+var UserConstants = require('col-users/lib/constants');
+
+/**
+ * Run through all the active courses and send out the daily notification email
+ *
+ * @param  {Function}     callback    Standard callback function
+ */
+var collect = module.exports.collect = function(callback) {
+  var start = Date.now();
+
+  // 1. Get all the active courses
+  CourseAPI.getCourses(false, function(err, courses) {
+    if (err) {
+        return log.error({'err': err}, 'Unable to retrieve the courses when sending notifications');
+    }
+
+    // 2. Iterate through all the courses
+    async.eachSeries(courses, collectCourse, function(err) {
+      if (err) {
+          return log.error({'err': err}, 'Unable to collect a course when sending notifications');
+      }
+
+      log.info('Sent emails for all courses, took %d ms', Date.now() - start);
```

Comment:
It looks like we mostly do the logging along the following lines:

```
log.info({'duration': (Date.now() - start)}, 'Sent emails for all courses');
```

So it might be worth doing that here as well